### PR TITLE
keyboard date input

### PR DESCRIPTION
**Pull request by penk85**
WPC-312, WPC-313, WPC-314: Update publish date and time manually.

keyboardpicker

### CODE REVIEW

**Review by HanaAyman — DISMISSED** [review]

File: packages/editor/src/client/atoms/dateTimePicker.tsx [hunk 1]
```diff
@@ -1,119 +1,107 @@
...
-  dateRanges?: DateTimeRange[]
-  timeRanges?: DateTimeRange[]
-
-  changeDate(publishDate: Date): void
+  dateRanges?: DateTimePreset[]
+  timeRanges?: DateTimePreset[]
 }
 
-export function DateTimePicker({
+export function KeyboardDateTimePicker({
```
Comment (bishoymelek) [inline packages/editor/src/client/atoms/dateTimePicker.tsx:24]: @penk85 Why did you renamed it?

**Review by bishoymelek — COMMENTED** [review]
Great Job @penk85 , some simple questions.

File: packages/editor/src/client/atoms/dateTimePicker.tsx [hunk 2]
```diff
@@ -1,119 +1,107 @@
...
-  changeDate
-}: DateTimePickerProps) {
+  timeRanges
+}: KeyboardDateTimePickerProps) {
   const {t} = useTranslation()
 
-  const date = new Date(dateTime?.getTime() ?? new Date())
+  const initialDate = new Date(dateTime ?? new Date())
+
+  const [startDate, setStartDate] = useState<any>(initialDate)
```
[the remark this answers is not in the document]
Comment (penk85) [reply packages/editor/src/client/atoms/dateTimePicker.tsx:35]: @bishoymelek `startDate` is the name for the selected date.. There's no end date because we aren't using ranges. I'll clean this up. :+1:

File: packages/editor/src/client/atoms/dateTimePicker.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (penk85) [reply packages/editor/src/client/atoms/dateTimePicker.tsx:24]: @bishoymelek the original component was from Rsuite. I changed the name to distinguish between them.

File: packages/editor/src/client/atoms/dateTimePicker.tsx [hunk 3]
```diff
@@ -1,119 +1,113 @@
...
   label: string
-  value: Date
+  offset: number
 }
 
-export interface DateTimePickerProps {
+export interface KeyboardDateTimePickerProps {
   dateTime: Date | undefined
   label: string
+  changeDate(publishDate: any): void
```
Comment (MostafaSalama) [inline packages/editor/src/client/atoms/dateTimePicker.tsx:18]: why we use `any` here? can we change it to the proper type?

Comment (penk85) [reply packages/editor/src/client/atoms/dateTimePicker.tsx:18]: As best I understand it can't just be a Date type because that function allows any keyboard input. I haven't found how to define the type here in a more specific way that won't also cause type errors in other places.

Comment (roosnic1) [reply packages/editor/src/client/atoms/dateTimePicker.tsx:18]: but if it can be a date or keyboard input then you could set the type to `Date | string`.

**Review by roosnic1 — CHANGES_REQUESTED** [review]
looks good just small change.

File: packages/editor/src/client/i18n.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import i18n from 'i18next'
 import {initReactI18next} from 'react-i18next'
 import LanguageDetector from 'i18next-browser-languagedetector'
+import {registerLocale, setDefaultLocale} from 'react-datepicker'
+
+import dateFnsDe from 'date-fns/locale/de'
```
Comment (bishoymelek) [inline packages/editor/src/client/i18n.ts:6]: Will we be using both date-fns and moment? @penk85 @roosnic1

**Review by bishoymelek — COMMENTED** [review]
Nice work @penk85 🎉 just a simple question

File: packages/editor/src/client/i18n.ts (hunk 4 above)
Comment (roosnic1) [reply packages/editor/src/client/i18n.ts:6]: @bishoymelek in #358 @camilleory already removes moment.js

**Review by camilleory — DISMISSED** [review]

**Review by lakeside763 — APPROVED** [review]
Great job, this looks good

File: packages/editor/src/client/i18n.ts [hunk 5]
```diff
@@ -40,12 +37,4 @@ i18n.use(LanguageDetector).use(initReactI18next).init({
   }
 })
 
-// TODO: how to handle other date amd time formats
-registerLocale('de', dateFnsDe)
```
Comment (roosnic1) [inline packages/editor/src/client/i18n.ts]: @penk85 why did you remove this?

File: packages/editor/src/client/i18n.ts [hunk 6]
```diff
@@ -1,9 +1,6 @@
 import i18n from 'i18next'
 import {initReactI18next} from 'react-i18next'
 import LanguageDetector from 'i18next-browser-languagedetector'
-import {registerLocale, setDefaultLocale} from 'react-datepicker'
-
-import dateFnsDe from 'date-fns/locale/de'
```
Comment (roosnic1) [inline packages/editor/src/client/i18n.ts]: I'm confused don't you need this.

Comment (bishoymelek) [reply packages/editor/src/client/i18n.ts]: @roosnic1 i think this one is outdated, I can't find it in the `files changed`

**Review by bishoymelek — APPROVED** [review]
🎉

File: packages/editor/src/client/i18n.ts (hunk 5 above)
Comment (penk85) [reply packages/editor/src/client/i18n.ts]: @roosnic1 was a merge conflict and I went with master version

File: packages/editor/src/client/i18n.ts (hunk 6 above)
Comment (penk85) [reply packages/editor/src/client/i18n.ts]: @roosnic1 also from merge

**Review by camilleory — DISMISSED** [review]

**Review by roosnic1 — DISMISSED** [review]

File: packages/editor/src/client/i18n.ts (hunk 6 above)
Comment (roosnic1) [reply packages/editor/src/client/i18n.ts]: Same here

**Review by roosnic1 — APPROVED** [review]
